Sources/NIO/NonBlockingFileIO.swift

```diff
@@ -277,4 +277,22 @@ public struct NonBlockingFileIO {
         }
     }
 
+    /// Open the file at `path` for writing on a private thread pool which is separate from any `EventLoop` thread.
+    ///
+    /// This function will return (a future) of the `NIOFileHandle` associated with the file opened and a `FileRegion`
+    /// comprising of the whole file. The caller must close the returned `NIOFileHandle` when it's no longer needed.
+    /// If file does not exist, it will be created with permissions 0644
+    ///
+    /// - note: The reason this returns the `NIOFileHandle` and the `FileRegion` is that both the opening of a file as well as the querying of its size are blocking.
+    ///
+    /// - parameters:
+    ///     - forWritingAtPath: The path of the file to be opened for writing.
+    ///     - eventLoop: The `EventLoop` on which the returned `EventLoopFuture` will fire.
+    /// - returns: An `EventLoopFuture` containing the `NIOFileHandle` and the `FileRegion` comprising the whole file.
+    public func openFile(forWritingAtPath path: String, eventLoop: EventLoop) -> EventLoopFuture<NIOFileHandle> {
```

Comment:
same comments here.

Reply:
What about returning `FileRegion` when opening for writing? If we have single API with an enum/option set, then we have to return `FileRegion`

Reply:
@artemredkin you're right, `FileRegion` probably doesn't make sense for writable stuff.

Reply:
then maybe we should have an API that supports read, write, read+write and does _not_ return a `FileRegion`

Reply:
Added another `openFile` method that requires `mode` and does not return a `FileRegion`